zebra-chain/src/transaction/serialize.rs

```diff
@@ -358,8 +360,8 @@ impl ZcashSerialize for Output {
 impl ZcashDeserialize for Output {
     fn zcash_deserialize<R: io::Read>(mut reader: R) -> Result<Self, SerializationError> {
         Ok(Output {
-            cv: reader.read_32_bytes()?,
-            cmu: reader.read_32_bytes()?,
+            cv: reader.read_32_bytes()?.into(),
+            cm_u: jubjub::Fq::from_bytes(&reader.read_32_bytes()?).unwrap(),
```

Comment:
This `unwrap` looks suspicious. The spec requires cm_u to be canonical as a consensus rule, but a non-canonical encoding can certainly occur.

Reply:
This should panic on `unwrap` if the encoding is non-canonical:
https://docs.rs/jubjub/0.3.0/jubjub/struct.Fq.html#method.from_bytes
![image](https://user-images.githubusercontent.com/552961/89703646-93c8aa00-d91a-11ea-98ad-f10bf8c86ece.png)

Reply:
But you don't want a panic here, do you? You want a `SerializationError`.

Reply:
Sure, either way we'll never allow a non-canonical scalar to deserialize

Reply:
Right, but I think the concern is that getting a malformed output description shouldn't DoS the node.

Reply:
This should be resolved in https://github.com/ZcashFoundation/zebra/pull/596/commits/b99a5508a339e693b5665d166b4048a2466b3fcd